lib/AST/Type.cpp

```diff
@@ -4610,3 +4637,42 @@ Type TypeBase::openAnyExistentialType(OpenedArchetypeType *&opened) {
   opened = OpenedArchetypeType::get(this);
   return opened;
 }
+
+bool TypeBase::hasOpaqueArchetypePropertiesOrCases() {
+  if (hasOpaqueArchetype())
+    return true;
```

Comment:
A type could be parameterized by an opaque archetype, but not use that type parameter in any of its fields or cases. Was this intentional?

Reply:
Yes. But I admit it is confusing. I will remove this hasOpaqueArchetype check in TypeBase and move it to the users of this API.

The place in TypeLowering that uses this predicate wants to know about either. It is used to map different type expansion context type lowerings to one map entry if we know that a type does not contain any opaque archetypes and is not resilient in any resilience domain.